fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1155,6 +1155,67 @@ void determineCommittedTransactions(CommitBatchContext* self) {
 	}
 }
 
+inline int64_t extractTenantIdFromStringRef(StringRef s) {
+	if (s.size() < TenantAPI::PREFIX_SIZE) {
+		return TenantInfo::INVALID_TENANT;
+	}
+	// Parse mutation key to determine mutation encryption domain
+	StringRef prefix = s.substr(0, TenantAPI::PREFIX_SIZE);
+	return TenantAPI::prefixToId(prefix, EnforceValidTenantId::False);
+}
+
+int64_t extractTenantIdFromSingleKeyMutation(MutationRef m) {
+	ASSERT(!isSystemKey(m.param1));
+	ASSERT(isSingleKeyMutation((MutationRef::Type)m.type));
+
+	return extractTenantIdFromStringRef(m.param1);
```

Comment:
I think this needs to handle the case where a versionstamped key is putting the versionstamp somewhere in the first 8 bytes as well (this should return an invalid tenant).